private/bufpkg/bufmodule/bufmodulecache/cas_module_cacher.go

```diff
@@ -93,36 +97,28 @@ func (c *casModuleCacher) PutModule(
 	modulePin bufmoduleref.ModulePin,
 	module bufmodule.Module,
 ) (retErr error) {
-	moduleManifest := module.Manifest()
-	if moduleManifest == nil {
-		return fmt.Errorf("manifest must be non-nil")
+	fileSet := module.FileSet()
+	if fileSet == nil {
+		return fmt.Errorf("FileSet must be non-nil")
 	}
-	manifestBlob, err := moduleManifest.Blob()
+	manifest := fileSet.Manifest()
+	// TODO: what about empty modules? Need to handle empty Manifests in bufcas
+	manifestBlob, err := bufcas.ManifestToBlob(manifest)
 	if err != nil {
 		return err
 	}
 	manifestDigest := manifestBlob.Digest()
-	if manifestDigest == nil {
-		return errors.New("empty manifest digest")
-	}
 	if modulePinDigestEncoded := modulePin.Digest(); modulePinDigestEncoded != "" {
-		modulePinDigest, err := manifest.NewDigestFromString(modulePinDigestEncoded)
+		modulePinDigest, err := bufcas.NewDigestForString(modulePinDigestEncoded)
 		if err != nil {
 			return fmt.Errorf("invalid module pin digest %q: %w", modulePinDigestEncoded, err)
 		}
-		if !manifestDigest.Equal(*modulePinDigest) {
+		if bufcas.DigestEqual(manifestDigest, modulePinDigest) {
```

Comment:
```suggestion
		if !bufcas.DigestEqual(manifestDigest, modulePinDigest) {
```

This is the reason for the failing test.

Reply:
Dang, fixed.